src/main/java/com/example/commandtest/service/UniversityService.java

```diff
@@ -0,0 +1,61 @@
+package com.example.commandtest.service;
+
+import com.example.commandtest.model.Department;
+import com.example.commandtest.repository.DepartmentRepository;
+import jakarta.persistence.EntityNotFoundException;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+@Service
+@RequiredArgsConstructor
+public class UniversityService {
+
+    private final DepartmentRepository departmentRepository;
+    private static final String DEPARTMENT_NOT_FOUND = "Not found department with name: ";
+
+    public String getDepartmentHead(String name) {
+        Department department = getDepartmentByName(name);
+        return department.getDepartmentHead().getFirstName() + " " + department.getDepartmentHead().getLastName();
```

Comment:
```suggestion
       return department.getDepartmentHead().asString();
```

or toString(). 

Just think of using 
```department.getDepartmentHead().getFirstName() + " " + department.getDepartmentHead().getLastName();```
for 100 times and then someone asks you to add few fields. And then again and again. 

Use asString() or toString() and you will change only in one place